gui/checkthread.cpp

```diff
@@ -197,7 +197,7 @@ void CheckThread::runAddonsAndTools(const ImportProject::FileSettings *fileSetti
                 QFile f1(analyzerInfoFile + '.' + addon + "-E");
                 if (f1.open(QIODevice::ReadOnly | QIODevice::Text)) {
                     QTextStream in1(&f1);
-                    const quint16 oldchksum = in1.readAll().toInt();
+                    const quint16 oldchksum = static_cast<quint16>(in1.readAll().toUInt());
```

Comment:
I don't like the compiler conversion warnings. I have disabled those in our Makefile/etc. I believe it's dangerous to fix these. The compilers are too sloppy.

how did you compile? Is there a missing -Wno-conversion or something in some project file?

Reply:
I'm using gcc 9.1.1 at the moment.
I'm going back to main branch now to recompile against both clang (ver 8) and gcc to double check findings. Building is done using CMake with a couple of options (BUILD_TESTS and BUILD_GUI). Nothing fancy here.

Reply:
Build Log
Clang - https://pastebin.com/mikvFjXV
GCC - https://pastebin.com/4AaXTSvk

Reply:
Build Log - post patch
Clang - https://pastebin.com/Diw6yNke
GCC - https://pastebin.com/dwceLs3U